server/src/service.rs

```diff
@@ -114,16 +114,61 @@ impl Service {
     }
 
     #[zbus(out_args("collection"))]
-    pub async fn read_alias(&self, _name: &str) -> Result<ObjectPath, ServiceError> {
-        todo!()
+    pub async fn read_alias(
+        &self,
+        name: &str,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<ObjectPath, ServiceError> {
+        let mut objectpath = ObjectPath::default();
+        for collection in self.collections.lock().await.iter() {
+            let collection_ifce_ref = object_server.interface::<_, Collection>(collection).await?;
+            let collection = collection_ifce_ref.get_mut().await;
+
+            if collection.alias().await == name || collection.label().await == name {
+                objectpath = ObjectPath::from(collection.path().clone());
+            }
+        }
+
+        Ok(objectpath)
     }
 
     pub async fn set_alias(
         &self,
-        _name: &str,
-        _collection: ObjectPath<'_>,
+        name: &str,
+        collection: ObjectPath<'_>,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
     ) -> Result<(), ServiceError> {
-        todo!()
+        if name != "default" {
+            return Err(ServiceError::ZBus(zbus::Error::Failure(String::from(
+                "Only the 'default' alias is supported",
+            ))));
+        }
+
+        let collections = self.collections.lock().await;
+
+        if !collections
```

Comment:
This check here is useless, instead, if you iterate the collections, make it return Ok(() the moment you found the corresponding collection, you can return the error at the end of the function as you won't reach it unless the collection doesn't exists

Reply:
Yeah, I agree. Thanks!